src/equations/linearized_euler_2d.jl

```diff
@@ -0,0 +1,174 @@
+    v1_prime = sinpi(2*t) * cospi(2*x[1])
+    v2_prime = sinpi(2*t) * cospi(2*x[2])
+    p_prime = rho_prime
+      
+    return SVector(rho_prime, v1_prime, v2_prime, p_prime)
+end
+    
+    
+"""
+    boundary_condition_wall(u_inner, orientation, direction, x, t, surface_flux_function,
+                                equations::LinearizedEulerEquations2D)
+    
+Boundary conditions for a solid wall.
+"""
+function boundary_condition_wall(u_inner, orientation, direction, x, t, surface_flux_function,
+                                    equations::LinearizedEulerEquations2D)
+    # Boundary state is equal to the inner state except for the velocity. For boundaries
+    # in the -x/+x direction, we multiply the velocity in the x direction by -1.
+    # Similarly, for boundaries in the -y/+y direction, we multiply the velocity in the
+    # y direction by -1
+    if direction in (1, 2) # x direction
+        u_boundary = SVector(u_inner[1], -u_inner[2], u_inner[3], u_inner[4])
+    else # y direction
+        u_boundary = SVector(u_inner[1], u_inner[2], -u_inner[3], u_inner[4])
+    end
+      
+    # Calculate boundary flux
+    if iseven(direction) # u_inner is "left" of boundary, u_boundary is "right" of boundary
+        flux = surface_flux_function(u_inner, u_boundary, orientation, equations)
+    else # u_boundary is "left" of boundary, u_inner is "right" of boundary
+        flux = surface_flux_function(u_boundary, u_inner, orientation, equations)
+    end
+      
+    return flux
+end
+    
+    
+# Calculate 1D flux for a single point
+@inline function flux(u, orientation::Integer, equations::LinearizedEulerEquations2D)
+    (; rho_mean, v1_mean, v2_mean, c) = equations
+    rho_prime, v1_prime, v2_prime, p_prime = u
+    if orientation == 1
+        f1 = v1_mean * rho_prime + rho_mean * v1_prime
+        f2 = v1_mean * v1_prime + p_prime / rho_mean
+        f3 = v1_mean * v2_prime
+        f4 = v1_mean * p_prime + c^2 * rho_mean * v1_prime
+    else
+        f1 = v2_mean * rho_prime + rho_mean * v2_prime
+        f2 = v2_mean * v1_prime
+        f3 = v2_mean * v2_prime + p_prime / rho_mean
+        f4 = v2_mean * p_prime + c^2 * rho_mean * v2_prime
+    end
+      
+    return SVector(f1, f2, f3, f4)
+end
+    
+    
+@inline have_constant_speed(::LinearizedEulerEquations2D) = Val(true)
+    
+@inline function max_abs_speeds(equations::LinearizedEulerEquations2D)
+    (; v1_mean, v2_mean, c) = equations
+    return abs(v1_mean) + c, abs(v2_mean) + c
+end
+    
+@inline function max_abs_speed_naive(u_ll, u_rr, orientation::Integer, equations::LinearizedEulerEquations2D)
+    (; v1_mean, v2_mean, c) = equations
+    if orientation == 1
+        return abs(v1_mean) + c
+    else # orientation == 2
+        return abs(v2_mean) + c
+    end
+end
+    
+    
+# Convert conservative variables to primitive
+@inline cons2prim(u, equations::LinearizedEulerEquations2D) = cons2cons(u, equations)
```

Comment:
```suggestion
@inline cons2prim(u, equations::LinearizedEulerEquations2D) = u
```
No need for going through `cons2cons`, is there?